source/guides/maintenance/custom-app-container.rst

```diff
@@ -0,0 +1,432 @@
+.. _custom-app-container:
+
+Modify existing application container images
+############################################
+
+This guide describes how to customize |CL-ATTR| application container images.
+
+.. contents::
+   :local:
+   :depth: 2
+
+Overview
+********
+
+|CL| publishes custom `Docker images on Docker Hub`_ of popular applications
+using |CL| as the base. Most of these images utilize a Docker build feature
+called a `multi-stage build to reduce image size`_ while some use
+single-stage build Dockerfiles. An official base
+`clearlinux image on Docker Hub`_ is also available. To create a generic |CL|
+container image, see :ref:`our guide <custom-clear-container>`.
+
+Prerequisites
+*************
+
+* Set up a functional Docker environment as described in :ref:`docker`.
+* Download the |CL| microservice Dockerfile repo with the following
+  command:
+
+  .. code-block:: bash
+
+     git clone https://github.com/clearlinux/dockerfiles.git
+
+Examples
+********
+
+Example 1: Add a bundle
+=======================
+
+In this example, we add :command:`wget` to the clearlinux/redis
+Dockerfile.
+
+#. Open a Terminal on |CL| and enter the :file:`dockerfiles` directory.
+
+#. Enter :command:`swupd search wget` to discover which |CL| bundle includes
+   the software.
+
+#. Open a an editor to modify the Dockerfile.
+
+#. Append the bundle to the `--bundles` parameter of the `swupd os-install`
+   command.
+
+#. Add :command:`wget` to the swupd command in the Dockerfile.
+
+#. Enter :command:`git diff` to show the edits made after
+   adding :command:`wget` in the clearlinux/redis Dockerfile.
+   The output shows:
+
+   .. code-block:: diff
+
+      diff --git a/redis/Dockerfile b/redis/Dockerfile
+      index af977cb..b1effab 100644
+      --- a/redis/Dockerfile
+      +++ b/redis/Dockerfile
+      @@ -15,7 +15,7 @@ RUN source /os-release && \
+          mkdir /install_root \
+          && swupd os-install -V ${VERSION_ID} \
+          --path /install_root --statedir /swupd-state \
+       -    --bundles=redis-native,findutils,su-exec --no-boot-update
+       +    --bundles=redis-native,findutils,su-exec,wget --no-boot-update
+
+       # For some Host OS configuration with redirect_dir on,

```

Comment:
```suggestion

```

Reply:
These two lines are technically part of the output but not relevant to the point so they can be removed for conciseness